src/test/java/uk/gov/ch/oauth/tokens/SessionSignInModifierTest.java

```diff
@@ -0,0 +1,117 @@
+package uk.gov.ch.oauth.tokens;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionKeys;
+
+@ExtendWith(MockitoExtension.class)
+class SessionSignInModifierTest {
+
+    @Spy
+    SessionSignInModifier sessionSignInModifier;
+    @Mock
+    private Session mockSession;
+    @Mock
+    private OAuthToken mockOAuthToken;
+    @Mock
+    private UserProfileResponse mockUserResponse;
+    @Mock
+    private Map<String, Object> sessionMap;
+    @Mock
+    private Map<String, Object> mockOAuthTokenMap;
+    @Mock
+    private Map<String, Object> mockUserResponseMap;
+
+    @Test
+    void alterSessionData_withObjects_delegatesWithCorrectMaps() {
+        when(mockSession.getData()).thenReturn(sessionMap);
+        when(mockOAuthToken.getAccessTokenAsMap()).thenReturn(mockOAuthTokenMap);
+        when(mockUserResponse.getUserProfileAsMap()).thenReturn(mockUserResponseMap);
+
+        //Guys want a second opinion on whether this is good or bad practice
+        doAnswer(assertNewMapIsCorrect()).when(sessionSignInModifier)
```

Comment:
This is confusing and hard to follow. What is the purpose of this?

Reply:
Might be better to call.
As this method doesn't return anything and generates a new object which needs appropriate values, this catches any time that there is a call that matches this address and checks that the second argument (newMap) is correct.